pomerium/gateway/backendrefs.go

```diff
@@ -0,0 +1,79 @@
+package gateway
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	gateway_v1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/pomerium/ingress-controller/model"
+	pb "github.com/pomerium/pomerium/pkg/grpc/config"
+)
+
+// applyBackendRefs translates backendRefs to a weighted set of Pomerium "To" URLs.
+// [applyFilters] must be called prior to this method.
+func applyBackendRefs(
+	route *pb.Route,
+	gc *model.GatewayHTTPRouteConfig,
+	backendRefs []gateway_v1.HTTPBackendRef,
+) {
+	for i := range backendRefs {
+		if !gc.ValidBackendRefs.Valid(gc.HTTPRoute, &backendRefs[i].BackendRef) {
+			log.Printf("backendRef %v not valid", &backendRefs[i].BackendRef) // XXX
```

Comment:
you're using regular logger here, need use `"sigs.k8s.io/controller-runtime/pkg/log"`

```
logger := log.FromContext(ctx)
```

Reply:
Good catch, I had put in some debug logging and missed this one. I'll take it out for now.

(I think probably if we log this it should come earlier, when we process an HTTPRoute itself.)